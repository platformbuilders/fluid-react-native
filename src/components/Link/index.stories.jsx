import React from 'react';
import { storiesOf } from '@storybook/react-native';
import { withKnobs, text, boolean, select } from '@storybook/addon-knobs';
import { action } from '@storybook/addon-actions';
import linkMarkdown from './link.md';
import Link from './index';

const optionsVariants = [
  'largeTitle',
  'title1',
  'title2',
  'title3',
  'title4',
  'headline',
  'body',
  'subhead',
  'footnote',
  'caption1',
  'caption2',
];

storiesOf('Link', module)
  .addDecorator(withKnobs)
  .addParameters({
    readme: {
      content: linkMarkdown,
    },
    viewport: {
      defaultViewport: '',
    },
  })
  .add('Example', () => (
    <Link
      variant={select('variant', optionsVariants, 'title1')}
      disabled={boolean('disabled', false)}
      onPress={action('link-click')}
      id="text _platform_builders"
      accessibility="testes"
      testID="id_text_platform_builders"
    >
      {text('children', 'Text Link')}
    </Link>
  ));
