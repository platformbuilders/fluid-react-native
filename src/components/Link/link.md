
# Link 

<!-- STORY -->

Source Code

<!-- SOURCE -->

<!-- PROPS -->
