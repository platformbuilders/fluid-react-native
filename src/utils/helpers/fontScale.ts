import { RFValue } from 'react-native-responsive-fontsize';

export const scaledFontSize = (
  size: number | string,
  baselineFactor?: 280,
): number => {
  const value = typeof size === 'string' ? parseInt(size, 10) : size;
  return RFValue(value, baselineFactor);
};
