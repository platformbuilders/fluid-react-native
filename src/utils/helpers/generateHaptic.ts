import Haptic, { HapticFeedbackType } from 'react-native-haptic';
import { isIOS } from './isIOS';

export const generateHaptic = (haptic: HapticFeedbackType = 'impact'): void => {
  if (isIOS()) {
    Haptic.generate(haptic);
  }
};
