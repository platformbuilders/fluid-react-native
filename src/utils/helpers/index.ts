export {
  currencyParser,
  parseToThousands,
  removeWhiteSpaces,
  toOnlyNumbers,
  getTheme,
  ifStyle,
  switchStyle,
} from '@platformbuilders/helpers';
export * from './generateHaptic';
export * from './getByTypographyVariant';
export * from './fontScale';
export * from './shadow';
