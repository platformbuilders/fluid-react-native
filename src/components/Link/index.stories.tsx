import { StyleProp, TextStyle } from 'react-native';
import type { Meta } from '@storybook/react';
import { VARIANT_SIZE } from '../../enums/size.enum';
import Link from '.';

const meta: Meta<typeof Link> = {
  title: 'Link',
  component: Link,
  argTypes: {
    onPress: { action: 'clicked' },
    variant: { control: { type: 'select' }, options: VARIANT_SIZE },
  },
  args: {
    accessibility: 'Link',
    accessibilityLabel: 'Link',
    children: 'Texto de exemplo',
    disabled: false,
    onPress: () => {},
    id: 'link-id',
    style: {
      color: '#355deb',
    } as StyleProp<TextStyle>,
    testID: 'link-test-id',
    variant: 'lg',
  },
  parameters: {
    notes:
      'Esse componente tem como função auxiliar na manipulação, manutenção e padronização de links dentro de uma aplicação.',
  },
};

export default meta;

export const Default = {};
