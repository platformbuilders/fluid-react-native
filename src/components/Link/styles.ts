import styled from 'styled-components/native';
import { getTheme } from '../../utils/helpers';
import Typography from '../Typography';

const primaryContrast = getTheme('primary.contrast');

type Props = {
  testID: string;
  accessibilityLabel: string;
  style: any;
};

export const Text = styled(Typography)<Props>`
  font-weight: 500;
  color: ${primaryContrast};
  text-decoration-color: ${primaryContrast};
  text-decoration-line: underline;
`;
