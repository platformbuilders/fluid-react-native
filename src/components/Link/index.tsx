import React, { FC } from 'react';
import { StyleProp, ViewStyle } from 'react-native';
import { TypographyVariants, TouchableType } from '../../types';
import { Text } from './styles';
import Touchable from '../Touchable';

interface Props extends TouchableType {
  variant?: TypographyVariants;
  style?: StyleProp<ViewStyle>;
}

const Link: FC<Props> = ({
  id,
  onPress,
  children,
  accessibility,
  accessibilityLabel,
  testID,
  variant = 'body',
  style,
  ...rest
}) => (
  <Touchable
    id={id || accessibility}
    onPress={onPress}
    accessibility={accessibility}
    accessibilityLabel={accessibilityLabel || accessibility}
    testID={testID || id || accessibility}
    {...rest}
  >
    <Text
      testID={`text_${id}`}
      accessibilityLabel={`Texto ${accessibility}`}
      accessibility={`Texto ${accessibility}`}
      style={style}
      variant={variant}
    >
      {children}
    </Text>
  </Touchable>
);

export default Link;
