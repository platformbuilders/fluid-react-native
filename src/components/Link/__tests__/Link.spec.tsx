/* eslint-disable jsx-a11y/anchor-is-valid */
import 'jest';
import React from 'react';
import renderer from 'react-test-renderer';
import { render, fireEvent } from 'react-native-testing-library';
import { ThemeProvider } from 'styled-components/native';
import Link from '..';
import { theme } from '../../../test/helpers';

describe('<Link />', () => {
  it('should render link', () => {
    const wrapper = renderer.create(
      <ThemeProvider theme={theme}>
        <Link id="testing" accessibility="" />
      </ThemeProvider>,
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  it('should trigger onPress function', () => {
    const onPressEvent = jest.fn();

    const { getByTestId } = render(
      <ThemeProvider theme={theme}>
        <Link id="testing" accessibility="" onPress={onPressEvent} />
      </ThemeProvider>,
    );

    const component = getByTestId('testing');

    fireEvent.press(component);

    expect(onPressEvent).toHaveBeenCalled();
  });
});
