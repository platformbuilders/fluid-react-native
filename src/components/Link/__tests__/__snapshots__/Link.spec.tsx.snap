// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Link /> should render link 1`] = `
<View
  accessibilityLabel=""
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID="testing"
>
  <Text
    accessibilityLabel="Texto "
    style={
      Array [
        Object {
          "color": "#212121",
          "fontSize": 31,
          "lineHeight": 43,
        },
        Object {
          "color": "#ffffff",
          "fontWeight": "500",
          "textDecorationColor": "#ffffff",
          "textDecorationLine": "underline",
        },
      ]
    }
    testID="text_testing"
    variant="body"
  />
</View>
`;
